submissions/dev-adventure/index.js

```diff
@@ -0,0 +1,77 @@
+const net = require('net');
+
+const startCheck = async () => {
+  try {
+    const arg = process.argv.slice(2);
+    if (!arg || !arg.length) {
+      process.stdout.write(
+        '--host <domain name or ip what you want to check>' +
+          ' is required arguments.\n' +
+          '--help for get description.'
+      );
+      process.exit(1);
+    }
+    if (arg[0] === '--help') {
+      process.stdout.write(
+        '--host <domain name or ip what you want to check> - required argument.\n' +
+          '--ports <start_port-end_port> - ports range from/to.\n' +
+          '--help - get help.'
+      );
+    }
+    let host;
+    let portStart;
+    let portEnd;
+    if (arg.length > 2) {
+      const ports = arg[1].split('-');
+      portStart = Number(ports[0]);
+      portEnd = Number(ports[1]);
+      [host] = arg.slice(3, 4);
+    } else {
+      [host] = arg.slice(0, 1);
+    }
+    const checkPort = port => {
+      return new Promise(resolve => {
+        let timer;
```

Comment:
Can you please explain your idea with timers?

Reply:
Yes, I can. When our timer is more than 300ms we close socket and set false status in resolve. So in this case we use timer like seconds count machine.